music21/metadata/__init__.py

```diff
@@ -1011,47 +1447,51 @@ def movementNumber(self) -> t.Optional[str]:
         >>> md.movementNumber
         '3'
         '''
-        result = self._workIds['movementNumber']
-        if result is not None:
-            return str(result)
-        return None
+        return self._getBackwardCompatibleItem('movementNumber')
 
     @movementNumber.setter
     def movementNumber(self, value):
-        self._workIds['movementNumber'] = Text(value)
+        self._setBackwardCompatibleItem('movementNumber', Text(value))
 
     @property
     def number(self):
         r'''
         Get or set the number of the work within a collection of pieces.
         (for instance, the number within a collection of ABC files)
 
+        >>> md = metadata.Metadata()
+        >>> md.number = 4
+
         Note that numbers are always returned as strings!  This may
         change in the future.
+
+        >>> md.number
+        '4'
         '''
-        result = self._workIds['number']
-        if result is not None:
-            return str(result)
+        return self._getBackwardCompatibleItem('number')
 
     @number.setter
     def number(self, value):
-        self._workIds['number'] = Text(value)
+        self._setBackwardCompatibleItem('number', Text(value))
 
     @property
     def opusNumber(self):
         r'''
         Get or set the opus number.
 
+        >>> md = metadata.Metadata()
+        >>> md.opusNumber = 56
+
         Note that opusNumbers are always returned as strings!  This may
         change in the future.
+        >>> md.opusNumber
+        '56'
```

Comment:
this one is unlikely to change in the future, actually, because of the number of opus like "56a" etc.